tools/snap/build_remote.py

```diff
@@ -28,21 +29,34 @@
 PLUGINS = [basename(path) for path in glob.glob(join(CERTBOT_DIR, 'certbot-dns-*'))]
 
 
+# In Python, stdout and stderr are buffered in each process by default. When
+# printing output from multiple processes, this can cause delays in printing
+# output with lines from different processes being interleaved depending
+# on when the output for that process is flushed. To prevent this, we override
+# print so that it always flushes its output. Disabling output buffering can
+# also be done through command line flags or environment variables set when the
+# Python process starts, but this approach was taken instead to ensure
+# consistent behavior regardless of how the script is invoked.
+print = functools.partial(print, flush=True)
```

Comment:
Alternatively, what do you think about modifying the shebang of this script into `#!/usr/bin/env python3 -u`?

Reply:
I actually started to do that initially but noticed calls to the script like `python3 tools/build_remote.py` on lines like https://github.com/certbot/certbot/blob/master/.azure-pipelines/templates/jobs/packaging-jobs.yml#L160. We could change those and use the shebang, but if we happened to call the script with `python` sometime in the future, it wouldn't work anymore.

Neither I or `pylint` love overriding builtins like this, but this is the approach I found I personally like best. I was a little surprised that there's not a great way to disable output buffering from inside Python once it has been started.

Reply:
Ok it is not a big deal for a tool script. Your approach will make it work in all cases.